pyocd/coresight/cortex_m.py

```diff
@@ -127,6 +139,21 @@ def register_name_to_index(reg):
             raise KeyError('cannot find %s core register' % reg)
     return reg
 
+# Returns true for registers holding single-precision float values
+def is_float_register(index):
+    return index >= 0x40 and index <= 0x5f
```

Comment:
Haha, hate to say it, but you opened the door… 😁 Could we have functions similar to `is_float_register()` for `is_cfbp_subregister()` and `is_xpsr_subregister()`? I'll do the same on the [dev_cm7](https://github.com/flit/pyOCD/tree/dev_cm7) branch with the double-precision float registers.

Eventually it might be good to change this so each register is represented by an object that knows how to perform the special logic associated with it, if any. I've never been too fond of these magic numbers.

Reply:
Yes, sure.